### ISSUE TITLE
Missing: "utils.visualization"

### ISSUE THREAD

**Issue by zer0int** [yossigandelsman/second_order_lens#1]
In the visualize_neurons notebook:
```
from utils.visualization import visualization_preprocess, image_grid
```
- No "visualization.py" in "utils".

PS: Thank you very much for this super insightful paper & code! I had some interesting results with my Geometric Parametrization (GmP) fine-tune (converted back to .weight thereafter) of ViT-L/14, Layer 20. 

![gmp-clip-results-L20](https://github.com/yossigandelsman/second_order_lens/assets/132047210/75d735a9-1fbc-40f9-bfc7-be2a27956e36)

I used an even smaller fraction of ImageNet than you did (I also have only 1 GPU available), so I'll have to check that against original pre-trained ViT-L/14 again with the same set; however, my GmP fine-tune outperforms pre-trained ViT-L/14 on ImageNet/ObjectNet, VOC2007_multilabel -- so the numbers are probably not too wrong (e.g. with the former, increase from accuracy ~0.84 -> 0.90). 

The lack of a drop in accuracy is surprising; however, the GmP fine-tune also happens to be much more robust against adversarial attacks in the form of typographic attacks (e.g. classic "apple ipod" example), so perhaps this is legitimate.
Either way - I am excited to see the results with regard to layer 21 I am computing right now.

Thanks again, I really appreciate your work!
PS: GmP code + link to fine-tuned model is on my github, just in case you're interested.

**Comment by yossigandelsman** [comment]
Thank you for detecting this bug and for sharing the GmP code! The bug will be fixed shortly!
